pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/DistinctCountCPCSketchAggregationFunction.java

```diff
@@ -419,6 +420,20 @@ public Comparable mergeFinalResult(Comparable finalResult1, Comparable finalResu
     return (Long) finalResult1 + (Long) finalResult2;
   }
 
+  @Override
+  public boolean canUseStarTree(Map<String, Object> functionParameters) {
+    Object lgK = functionParameters.get(Constants.CPCSKETCH_LGK_KEY);
+
+    // Check if lgK values match
+    if (lgK != null) {
+      return _lgNominalEntries == Integer.parseInt(String.valueOf(lgK));
+    } else {
+      // If the functionParameters don't have an explicit lgK set, it means that the star-tree index was built with
+      // the default value for lgK
+      return _lgNominalEntries == CommonConstants.Helix.DEFAULT_CPC_SKETCH_LGK;
```

Comment:
long nit:

Not super important, but I would recommend in the future to either use `long` instead of `Long` or use `equals` instead of `==`.

Remember that boxed integers have to be compared with `.equals` instead of `==` because `==` will use identity comparison. For example:

```jsell
jshell> Long a1234 = 1234L;
a1234 ==> 1234

jshell> Long aLarge = 1234L;
aLarge ==> 1234

jshell> a1234 == aLarge;
$3 ==> false

jshell> a1234.equals(aLarge);
$4 ==> true
```

In this case we don't have that problem because `CommonConstants.Helix.DEFAULT_CPC_SKETCH_LGK` is an `int` not an `Integer` so we are doing `Integer == int`, which is going to use value equality. But if for whatever reason we change `CommonConstants.Helix.DEFAULT_CPC_SKETCH_LGK` to `Integer`, this code will be incorrect without warning.

This is even more problematic because Integers are cached for small values, so:
```
jshell> Long a1 = 1L;
a1 ==> 1

jshell> Long aSmall = 1L;
aSmall ==> 1

jshell> a1 == aSmall;
$7 ==> true

jshell> a1.equals(aSmall);
$8 ==> true
```

Reply:
I'm not sure I follow - `_lgNominalEntries` here is also an `int` so we're doing `int == int`.

Reply:
You are right. I thought we were comparing `lgK`.